src/App.tsx

```diff
@@ -1,24 +1,207 @@
+import { Footer } from './components/Footer';
+import { FilterValues } from './types/FilterValues';
+import { ErrorValues } from './types/ErrorValues';
+import { Notification } from './components/Notification';
+import { ErrorContext } from './context/ErrorContextProvider';
 
-const USER_ID = 0;
+const USER_ID = 10544;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [filterValue, setFilterValue] = useState(FilterValues.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [completedTodosID, setCompletedTodosID] = useState<number[]>([]);
+  const [todosForTemp, setTodosForTemp] = useState<number[]>([]);
+  const errorContext = useContext(ErrorContext);
+  const inputHeaderRef = useRef() as MutableRefObject<HTMLInputElement>;
+  const leftTodosCount
+    = todosFromServer?.filter(todo => !todo.completed).length || 0;
+
+  const filteredTodos = (todos: Todo[], value: FilterValues) => {
+    return todos.filter(todo => {
+      switch (value as FilterValues) {
+        case FilterValues.Completed:
+          return todo.completed;
+        case FilterValues.Active:
+          return !todo.completed;
+        case FilterValues.All:
+        default:
+          return true;
+      }
+    });
+  };
+
+  const getId = (todos: Todo[]) => {
+    return todos.map(todo => todo.id);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(res => {
+        setTodosFromServer(res);
+      })
+      .catch(() => {
+        errorContext.setErrorMessage(ErrorValues.Loading);
+      });
+  }, []);
+
+  useEffect(() => {
+    setCompletedTodosID(() => getId(filteredTodos(todosFromServer,
+      FilterValues.Completed)));
+  }, [todosFromServer, leftTodosCount]);
+
+  const todosAfterFilter = useMemo(() => {
+    return filteredTodos(todosFromServer, filterValue);
+  }, [todosFromServer, filterValue]);
+
+  const handleAddingTodos = (inputValue: string) => {
+    const newTodo = {
+      userId: USER_ID,
+      title: inputValue,
+      completed: false,
+    };
+
+    setTempTodo({
+      id: 0,
+      ...newTodo,
+    });
+
+    addTodo(USER_ID, newTodo)
+      .then(res => setTodosFromServer(prevTodos => [...prevTodos, res]))
+      .catch(() => errorContext.setErrorMessage(ErrorValues.Adding))
+      .finally(() => {
+        inputHeaderRef.current.disabled = false;
+        inputHeaderRef.current.focus();
+        setTempTodo(null);
+      });
+  };
+
+  const handleUpdatingTodo = (id: number, data: null | string) => {
+    const todoForUpdating = todosFromServer.find(t => t.id === id);
+    let dataForUpdating = {};
+
+    if (todoForUpdating && data === null) {
+      dataForUpdating = todoForUpdating.completed
+        ? { completed: false }
+        : { completed: true };
+    } else if (todoForUpdating && typeof data === 'string') {
+      if (data.length) {
+        dataForUpdating = { title: data };
+      } else {
+        return;
```

Comment:
try to simplify this `if/else` block, because it's hard to read. you can just use `if` everywhere you need, and `return;` at the end of it if you want to stop the function

Reply:
Optimize conditions and make them more readable, example:

```
if (todoForUpdating) {
  if (!data) {
    dataForUpdating = { completed: !todoForUpdating.completed };
  } else if (typeof data === 'string' && data.length) {
    dataForUpdating = { title: data };
  }
}
```